Use F::*FuncOptions for embedding/embeddingbag functionals

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29673 Use F::*FuncOptions for embedding/embeddingbag functionals **
* #29584 Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h
* #29536 Use c10::variant-based enums for SmoothL1Loss module and functional
* #29535 Use c10::variant-based enums for F::grid_sample
* #29404 Make all non-input arguments to functionals part of its options, and pass functional options by const-ref when possible

Following https://github.com/pytorch/pytorch/pull/29364 and https://github.com/pytorch/pytorch/pull/29404, this PR makes `F::EmbeddingFuncOptions` and `F::EmbeddingBagFuncOptions` separate classes from `torch::nn::EmbeddingOptions` and `torch::nn::EmbeddingBagOptions`, so that it's easier to enforce that arguments such as `num_embeddings` and `embedding_dim` are required for `torch::nn::EmbeddingOptions` and `torch::nn::EmbeddingBagOptions`.

Differential Revision: [D18462540](https://our.internmc.facebook.com/intern/diff/D18462540)

Commit:
Use F::*FuncOptions for embedding/embeddingbag functionals

Commit:
Update on "Use F::*FuncOptions for embedding/embeddingbag functionals"

Use F::*FuncOptions for embedding/embeddingbag functionals

gh-metadata: pytorch pytorch 29673 gh/yf225/68/head

kostmo commented:
## CircleCI build failures summary
As of commit 0698505:
* **1/1** failures introduced in this PR
* **0/1** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=0698505e58bb287011d1b2fe686c645f24312df1).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

Commit:
Update on "Use F::*FuncOptions for embedding/embeddingbag functionals"

Use F::*FuncOptions for embedding/embeddingbag functionals

gh-metadata: pytorch pytorch 29673 gh/yf225/68/head
